browser/resources/settings/email_aliases_page/remote_mapping_service.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { AccountState, Alias, MappingService } from './types'
+import { sendWithPromise} from 'chrome://resources/js/cr.js';
+
+export class RemoteMappingService implements MappingService {
+  private pending_cancellation_ = false
+
+  async getAccountEmail (): Promise<string | undefined> {
+    return sendWithPromise('email_aliases.getAccountEmail')
+  }
+  async logout (): Promise<void> {
+    await sendWithPromise('email_aliases.logout')
+  }
+  async requestAccount (accountEmail: string): Promise<void> {
+    await sendWithPromise('email_aliases.requestAccount', accountEmail)
+  }
+  async getAccountState (): Promise<AccountState> {
+    throw new Error('Method not implemented.');
+  }
+  async getAliases (): Promise<Alias[]> {
+    const result = await sendWithPromise('email_aliases.getAliases')
+    return result
+  }
+  async createAlias (email: string, note: string): Promise<void> {
+    await sendWithPromise('email_aliases.createAlias', email, note)
+  }
+  async updateAlias (email: string, note: string, status: boolean): Promise<void> {
+    await sendWithPromise('email_aliases.updateAlias', email, note, status)
+  }
+  async deleteAlias (email: string): Promise<void> {
+    await sendWithPromise('email_aliases.deleteAlias', email)
+  }
```

Comment:
hey, this should be using a mojom interface, rather than a page handler

Reply:
Will be done in a follow-up: https://github.com/brave/brave-browser/issues/44806